File: services/user-service/src/main/java/ru/sg/user/event/UserRegisteredEvent.java
package ru.sg.user.event;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserRegisteredEvent {
    private String keycloakId;
    private String username;
    private String email;
    private String firstName;
    private String lastName;
    @JsonFormat(shape = JsonFormat.Shape.STRING)
    private Instant registeredAt;
    @JsonFormat(shape = JsonFormat.Shape.STRING)
    private Instant updatedAt;
}

File: services/notification-service/src/main/kotlin/ru/sg/email/dto/event/UserRegisteredEvent.kt
package ru.sg.email.dto.event

import java.time.Instant

data class UserRegisteredEvent(
    val keycloakId: String,
    val username: String,
    val email: String,
    val firstName: String,
    val lastName: String,
    val registeredAt: Instant,
    val updatedAt: Instant
)